[py] error: cast from 'void*' to 'long unsigned int' loses precision

All good for 32-bit architecture, but for 64-bit build we get compilation error.

For `MINGW64`:
```console
  C:/_/mingw-w64-partio/src/build-x86_64-w64-mingw32/src/py/CMakeFiles/partio_mod.dir/partioPYTHON_wrap.cxx: In function 'long unsigned int ParticlesDataMutable_ptr(const Partio::ParticlesDataMutable*)':
  C:/_/mingw-w64-partio/src/build-x86_64-w64-mingw32/src/py/CMakeFiles/partio_mod.dir/partioPYTHON_wrap.cxx:3841:93: error: cast from 'void*' to 'long unsigned int' loses precision [-fpermissive]
   3841 | SWIGINTERN unsigned long ParticlesDataMutable_ptr(ParticlesDataMutable const *self){ return (unsigned long)(void*)self; }
        |                                                                                             ^~~~~~~~~~~~~~~~~~~~~~~~~~
```

For `CLANG64`:
```console
  D:\a\_temp\msys64\clang64\bin\clang++.exe -DPARTIO_USE_ZLIB -D_USE_MATH_DEFINES -Dpartio_mod_EXPORTS -ID:/a/_temp/msys64/clang64/include/python3.9 -IC:/_/mingw-w64-partio/src/partio-1.14.6/src/lib -march=x86-64 -mtune=generic -O2 -pipe -O3 -DNDEBUG -pthread -std=c++14 -MD -MT src/py/CMakeFiles/partio_mod.dir/CMakeFiles/partio_mod.dir/partioPYTHON_wrap.cxx.obj -MF src\py\CMakeFiles\partio_mod.dir\CMakeFiles\partio_mod.dir\partioPYTHON_wrap.cxx.obj.d -o src/py/CMakeFiles/partio_mod.dir/CMakeFiles/partio_mod.dir/partioPYTHON_wrap.cxx.obj -c C:/_/mingw-w64-partio/src/build-x86_64-w64-mingw32/src/py/CMakeFiles/partio_mod.dir/partioPYTHON_wrap.cxx
  C:/_/mingw-w64-partio/src/build-x86_64-w64-mingw32/src/py/CMakeFiles/partio_mod.dir/partioPYTHON_wrap.cxx:3841:93: error: cast from pointer to smaller type 'unsigned long' loses information
  SWIGINTERN unsigned long ParticlesDataMutable_ptr(ParticlesDataMutable const *self){ return (unsigned long)(void*)self; }
                                                                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~
```

I use toolchains from [MSYS2](https://www.msys2.org/).

--- a/src/py/partio.i
+++ b/src/py/partio.i
@@ -465,7 +465,7 @@ public:
 
     %feature("autodoc");
     %feature("docstring","Workaround to get the address to the ptr to help with interop python binding");
-    unsigned long ptr() const { return (unsigned long)(void*)self; }
+    unsigned long long ptr() const { return (unsigned long long)(void*)self; }
 }
 
 %extend ParticlesInfo {

